### PR TITLE
J3RN's solution

### PR DESCRIPTION
**Pull request by J3RN**
I'd love some feedback on my homebrewed auth!

### CODE REVIEW

File: source/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,13 +2,23 @@ class ApplicationController < ActionController::Base
...
 
-  def authenticate_user
+  private
+
+  def logged_in?
+    session[:user_id].present?
+  end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```
Comment (jaybobo) [inline source/app/controllers/application_controller.rb:22]: `if logged_in?`

File: source/app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -4,9 +4,28 @@ def index
...
+      session[:user_id] = user.id
+      redirect_to user
+    else
+      flash.now[:alert] = "Email or password incorrect!"
+      render 'new'
+    end
   end
 
   def destroy
+    if session[:user_id]
```
Comment (jaybobo) [inline source/app/controllers/sessions_controller.rb]: if this inherits from ApplicationController, its methods should be accessible here.

**Comment by J3RN** [comment]
Updated!

**Comment by jaybobo** [comment]
:+1:

**Pull request closed without merging**